druid/src/core.rs

```diff
@@ -527,11 +550,14 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                 Event::Timer(_) => (),
                 Event::WindowConnected => (),
                 Event::WindowSize(_) => (),
+                // Should commands require layout?
+                // If yes, it will cause problems with our test suite.
+                Event::Command(_) => (),
                 _ => {
-                    log::warn!(
-                        "Widget '{}' received an event ({:?}) without having been laid out. \
```

Comment:
I had to exclude `Event::Command` here, because otherwise our tests would break.
There we often send test commands without doing any layout.

This can of course be changed back, but I wanted to push this now without meddling too much with our test stuff.

Reply:
This came up in #1160 recently, because unviewed tabs don't get laid out but they might still receive commands. I think there's a bigger decision that needs to be made (and documented) about which events and lifecycle events hidden widgets should expect to receive, but for now I think it's completely reasonable for them to get commands.

Reply:
I agree, it's even totally reasonable for a `Command` to be sent while handling `Event::WindowConnected`, and in that case layout should *not* have been called.

Reply:
There are methods in the tabs code that try to say whether an event or lifecycle should be delivered to hidden widgets. Possibly should be moved to the enums themselves.